src/yoroi-wallets/cardano/shelley/transactionCache.ts

```diff
@@ -505,6 +518,22 @@ export const makeTxCacheStorage = (storage: Storage): TxCacheStorage => ({
   },
 })
 
+const addMemos = async (transactions: Record<string, Transaction>, memosStorage: Storage) => {
+  const memosTxIds = await memosStorage.getAllKeys()
+
+  memosTxIds.forEach(async (memoTxId) => {
+    const memo = (await memosStorage.getItem(memoTxId)) as string
```

Comment:
`array.forEach` won't work with `async` and the `addMemos` may resolve before all the promises inside the function resolve. `const promises = memosTxIds.map( async (memotxId)) => ...` should work or a `for` loop should work too